Source/WebDriver/Session.cpp

```diff
@@ -3282,21 +3283,89 @@ void Session::emitEvent(const String& eventName, RefPtr<JSON::Object>&& body)
 bool Session::eventIsEnabled(const String& eventName, const Vector<String>&)
 {
     // https://w3c.github.io/webdriver-bidi/#event-is-enabled
-    HashSet<String> topLevelBrowsingContexts;
-    // FIXME Add support to subscribe to specific browsing contexts
-    // https://bugs.webkit.org/show_bug.cgi?id=282981
+    for (const auto& subscription : m_subscriptions) {
 
-    return m_globalEventSet.contains(eventName);
+        // FIXME Add support to subscribe to specific browsing contexts
+        // https://bugs.webkit.org/show_bug.cgi?id=282981
+        if (!subscription.value.isGlobal())
+            continue;
+
+        for (const auto& event : subscription.value.events) {
+            if (event == eventName)
+                return true;
+        }
+    }
+
+    return false;
+}
+
+void Session::subscribe(Vector<String>&& events, Vector<String>&& browsingContextIDs, Vector<String>&& userContextIDs, Function<void(CommandResult&&)>&& completionHandler)
+{
+    // TODO Process/validate list of event names (e.g. expanding if given only the module name)
+    // https://bugs.webkit.org/show_bug.cgi?id=291371
+    auto subscriptionID = WTF::createVersion4UUIDString();
+    Subscription subscription = { subscriptionID, WTFMove(events), WTFMove(browsingContextIDs), WTFMove(userContextIDs) };
+    m_subscriptions.add(subscriptionID, WTFMove(subscription));
```

Comment:
You might be able to inline the initializer for `subscription` since we just used the ID after this point.